### ISSUE TITLE
[BUG] Fails ungracefully if key and secret are not set

### ISSUE THREAD

**Issue by peterhoeg** [SamSchott/maestral-dropbox#40]
**Describe the bug**
When using maestral compiled from source (or packaged for a distro that doesn't include the standard keys), maestral will fail with an exception if ```DROPBOX_API_KEY``` and ```_SECRET``` are not set.

```
Traceback (most recent call last):
  File "/nix/store/jlbx4m9zv15b9hbf63s3wx2407l9h906-maestral-gui-0.2.6/lib/python3.7/site-packages/maestral/gui/setup_dialog.py", line 137, in on_link
    self.auth_url = self.auth_session.get_auth_url()
  File "/nix/store/jlbx4m9zv15b9hbf63s3wx2407l9h906-maestral-gui-0.2.6/lib/python3.7/site-packages/maestral/oauth.py", line 64, in get_auth_url
    APP_KEY = os.environ["DROPBOX_API_KEY"]
  File "/nix/store/w7gsq8v86hni4ynaqgwwlnlny115ylng-python3-3.7.4/lib/python3.7/os.py", line 678, in __getitem__
    raise KeyError(key) from None
KeyError: 'DROPBOX_API_KEY'
[1]    6765 abort      maestral-gui
```

**To Reproduce**
1. Compile from source.
2. Try to launch maestral without setting the required environment variables

**Expected behaviour**
A graceful exit and message about the required variables not being set.

**System (please complete the following information):**
 - OS: NixOS
 - Desktop environment: KDE
 - Qt version: 5.12

**Comment by SamSchott** [comment]
Yes, I know. This has been fixed in the development branch (and in v0.2.7-beta2) where Maestral no longer needs an API secret. 

I have changed the auth flow to OAuth2 Implicit Grant, following Dropbox's recommendation for client-side apps. This trades an exposed API secret, even when obfuscated in the binary, for slightly less secure auth flow.

**Comment by SamSchott** [comment]
Right, the newest version v0.3.0 no longer requires an APP_SECRET and has the APP_KEY hard coded.

For the future, I will switch to [PKCE for OAuth2](https://oauth.net/2/pkce/) if / when Dropbox supports it.
